src/umpire/strategy/ZeroByteHandler.cpp

```diff
@@ -35,11 +35,7 @@ void* ZeroByteHandler::allocate(std::size_t bytes)
 
 void ZeroByteHandler::deallocate(void* ptr)
 {
-  if (!m_zero_byte_pool)
-    m_zero_byte_pool = static_cast<FixedPool*>(
-        ResourceManager::getInstance().getZeroByteAllocator());
-
-  if (m_zero_byte_pool->pointerIsFromPool(ptr)) {
+  if (m_zero_byte_pool && m_zero_byte_pool->pointerIsFromPool(ptr)) {
```

Comment:
@davidbeckingsale, while this fix works, I think further examination of the FixedBlockPool manager may be necessary.  It appeared to me that the FixedBlockPool manager may have missed some initialization of some of the `m_pool` elements which caused `m_zero_byte_pool->pointerIsFromPool(ptr)` to return true even when the `m_pool` was created but had no allocations present.

Reply:
This change was not necessary.  It was caused by something else.